test/conformance/api/v1/util.go

```diff
@@ -121,26 +121,22 @@ func validateDataPlane(t testing.TB, clients *test.Clients, names test.ResourceN
 // Validates the state of Configuration, Revision, and Route objects for a runLatest Service.
 // The checks in this method should be able to be performed at any point in a
 // runLatest Service's lifecycle so long as the service is in a "Ready" state.
-func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string, imagesForMultipleContainers ...map[string]string) error {
+func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string) error {
 	t.Log("Checking to ensure Revision is in desired state with", "generation", expectedGeneration)
 	if err := v1test.CheckRevisionState(clients.ServingClient, names.Revision, func(r *v1.Revision) (bool, error) {
 		if ready, err := v1test.IsRevisionReady(r); !ready {
 			return false, fmt.Errorf("revision %s did not become ready to serve traffic: %w", names.Revision, err)
 		}
-		// for multi containers there will be more than one container and names.Image is a string,
-		// so in order to validate imageDigest for each images validateControlPlane accepting optional field called imagesForMultipleContainers
-		if names.Image == "" {
-			for _, value := range imagesForMultipleContainers {
-				for _, v := range r.Status.ContainerStatuses {
-					if image, ok := value[v.Name]; ok {
-						if validDigest, err := shared.ValidateImageDigest(t, image, v.ImageDigest); !validDigest {
-							return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", v.ImageDigest, image, err)
-						}
-					}
+		for i, v := range r.Status.ContainerStatuses {
+			if names.Image != "" {
+				if validDigest, err := shared.ValidateImageDigest(t, names.Image, v.ImageDigest); !validDigest {
+					return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", v.ImageDigest, names.Image, err)
+				}
+			} else {
+				if validDigest, err := shared.ValidateImageDigest(t, names.Sidecars[i], v.ImageDigest); !validDigest {
+					return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", v.ImageDigest, names.Sidecars[i], err)
```

Comment:
With my comment below in mind, I think this should rather be something like

```golang
// This at the top of the function somewhere.
images := append([]string{names.Image}, names.Sidecars...}

// Loop over them in a general way, without the if
```

Reply:
Modified